lizmap/modules/lizmap/lib/Request/RemoteStorageRequest.php

```diff
@@ -0,0 +1,361 @@
+
+                        return null;
+                    }
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Check if remote storage is reacheable on setted configuration.
+     *
+     * @return bool
+     */
+    public static function checkWebDAVStorageConnection()
+    {
+        $profile = self::getProfile('webdav');
+        if ($profile) {
+            $opt = array();
+            $headers = array();
+            $headers['User-Agent'] = 'lizmap-user-agent';
+
+            if (array_key_exists('user', $profile) && array_key_exists('password', $profile)) {
+                $opt['auth'] = array();
+                array_push($opt['auth'], $profile['user']);
+                array_push($opt['auth'], $profile['password']);
+            }
+            $opt['headers'] = $headers;
+
+            $client = new Client();
+
+            try {
+                $response = $client->request('GET', $profile['baseUri'], $opt);
+
+                return true;
+            } catch (\GuzzleHttp\Exception\RequestException $e) {
+                return false;
+            } catch (\Exception $e) {
+                return false;
+            }
+        }
+
+        return false;
+    }
+
+    public static function getAppContext()
+    {
+        if (!self::$appContext) {
+            self::$appContext = \lizmap::getAppContext();
+        }
+
+        return self::$appContext;
+    }
+
+    public static function getProfile($storageType, $profileName = 'default')
+    {
+        $context = self::getAppContext();
+        $profile = null;
+
+        try {
+            $profile = $context->getProfile($storageType, $profileName, true);
+            if ($profile && is_array($profile) && array_key_exists('enabled', $profile) && $profile['enabled'] == 1 && array_key_exists('baseUri', $profile)) {
+                return $profile;
+            }
+        } catch (\Exception $e) {
+            $profile = null;
+        }
+
+        return $profile;
+    }
+
+    /**
+     * Return the WebDav Url or null if fails
+     * The function assumes that the last part of the url is the filename and is defined as "file_name(@selected_file_path)".
+     *
+     * @param string      $storageUrl remote destination folder
+     * @param null|string $filename   file name, if null return the base path
+     *
+     * @return null|string
+     */
+    public static function getRemoteUrl($storageUrl, $filename = null)
+    {
+        $storeLen = strlen($storageUrl);
+        $fileNameLen = strlen(self::$fileNameExpression);
+        if ($fileNameLen < $storeLen && substr_compare($storageUrl, self::$fileNameExpression, $storeLen - $fileNameLen, $fileNameLen) === 0) {
+            if ($filename) {
+                // TODO @selected_file_path property is not evaluated, for now replace the expression with the file name
+                return str_replace('file_name(@selected_file_path)', "'".$filename."'", $storageUrl);
```

Comment:
`self::$fileNameExpression` is made for that, no ?
```suggestion
                return str_replace(self::$fileNameExpression, "'".$filename."'", $storageUrl);
```